src/act.offensive.cpp

```diff
@@ -598,7 +598,19 @@ bool passed_flee_success_check(struct char_data *ch) {
   if (!can_hurt(ch, FIGHTING(ch), TRUE, 0))
     return TRUE;
 
-  return success_test(GET_QUI(ch), GET_QUI(FIGHTING(ch))) > 0;
+  if success_test(GET_QUI(ch), GET_QUI(FIGHTING(ch)) + racial_flee_modifier) > 0
+    switch (GET_RACE(ch)) {
+      case RACE_SATYR:
+       racial_flee_modifier--;
```

Comment:
You're changing racial_flee_modifier after it's used in the success test, so this code won't do anything. The switch statement will need to come before the success_test call.